### DISCUSSION TITLE
Installing MDAlaysis+ MDAnalysisTests

### DISCUSSION THREAD

**Discussion by hanahg** [MDAnalysis/mdanalysis#4358] [Installation] [answered]
HI
I am installing MDAalysis and after following command I got the errors.
`mamba install -c conda-forge MDAnalysisTests` 
I tried to install the versions of python required but the problem did not solved yet.
Please help me to solve the problem.

Thank you

The anaconda version is `23.9.0`

 The error: 
```
looking for: ['mdanalysistests']

pkgs/r/linux-64                                               No change
pkgs/main/linux-64                                            No change
pkgs/main/noarch                                              No change
pkgs/r/noarch                                                 No change
bioconda/noarch                                               No change
bioconda/linux-64                                             No change
conda-forge/noarch                                  12.8MB @   5.5MB/s  2.7s
conda-forge/linux-64                               @   7.6MB/s  5.1s

Pinned packages:
  - python 3.12.*

warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
Could not solve for environment specs
The following packages are incompatible
└─ mdanalysistests is installable with the potential options
   ├─ mdanalysistests 0.15.0 would require
   │  └─ mdanalysis 0.15.0 , which does not exist (perhaps a missing channel);
   ├─ mdanalysistests 0.16.0 would require
   │  └─ mdanalysis 0.16.0 , which does not exist (perhaps a missing channel);
   ├─ mdanalysistests 0.16.1 would require
   │  └─ mdanalysis 0.16.1 , which does not exist (perhaps a missing channel);
   ├─ mdanalysistests [0.16.2|0.17.0|...|0.20.1] would require
   │  └─ python [2.7* |>=2.7,<2.8.0a0 ], which can be installed;
   ├─ mdanalysistests [0.17.0|0.18.0] would require
   │  └─ python [3.5* |>=3.5,<3.6.0a0 ], which can be installed;
   ├─ mdanalysistests 0.17.0 would require
   │  └─ python 3.6* , which can be installed;
   ├─ mdanalysistests [0.18.0|0.19.0|...|2.0.0] would require
   │  └─ python >=3.6,<3.7.0a0 , which can be installed;
   ├─ mdanalysistests [0.19.2|0.20.1|...|2.1.0] would require
   │  └─ python >=3.7,<3.8.0a0 , which can be installed;
   ├─ mdanalysistests [0.20.1|1.0.0|...|2.2.0] would require
   │  └─ python >=3.8,<3.9.0a0 , which can be installed;
   ├─ mdanalysistests [2.0.0|2.1.0|2.2.0] would require
   │  └─ python >=3.9,<3.10.0a0 , which can be installed;
   ├─ mdanalysistests [2.1.0|2.2.0] would require
   │  └─ python >=3.10,<3.11.0a0 , which can be installed;
   ├─ mdanalysistests [2.2.0|2.3.0] would require
   │  └─ python >=3.8,<=3.10 , which can be installed;
   ├─ mdanalysistests [2.4.1|2.4.2|2.4.3] would require
   │  └─ python >=3.8,<=3.11 , which can be installed;
   └─ mdanalysistests [2.5.0|2.6.1] would require
      └─ python >=3.9,<=3.11 , which can be installed.
```

**Comment by IAlibay** [comment, marked as answer]
@hanahg - am I correct in understanding that you are using Python 3.12?

Unfortunately MDAnalysis does not have a py3.12 compatible release yet, you would have to downgrade to py3.11.

**Comment by hanahg** [comment]
Thanks for your reply.
I didn't notice that. I will install the correct version of python.



Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by hanahg** [comment]
Hi,
I have reinstalled Anaconda and I have found that Python 3.12 is installed with this version of Anaconda3-2023.09. If not the Python version of my system is 3.11. I installed python 3.11 on Anaconda and again installed MDAnalysis. However, for this command I received this message "mamba install -c conda-forge mdanalysis
Currently, only install, create, list, search, run, info, clean, remove, update, repoquery, activate and deactivate are supported through mamba." and I don't know what should I do then.
I uninstalled all and I am trying by pip.

**Reply by RMeli** [reply]
Hi @hanahg, please have a look at @IAlibay suggestion below. You likely used the Anaconda installer which bundles Python `3.12`, and this will be the version of Python in your `base` environment. You can create a new environment with Python `3.11` as suggested below by @IAlibay. Using environments instead of installing everything in the `base` environment is strongly recommended.

**Reply by hanahg** [reply]
I did not find his discussion.
Could you please send the link?

Thanks

**Reply by IAlibay** [reply]
@hanahg it's just below on the page, here's the link just in case: https://github.com/MDAnalysis/mdanalysis/discussions/4358#discussioncomment-7790273

**Comment by IAlibay** [comment]
@hanahg Unfortunately it's unclear to me as to what specific steps you took to get this mamba error. May I suggest reaching out on discord where folks can more interactively respond to your questions?

The general approach with anaconda is to use environments which let you define a specific python version & packages.

You can see more about this here: https://userguide.mdanalysis.org/stable/installation.html#conda

Your typical usage pattern would be something like:

```
conda create --name mdanalysis python=3.11 MDAnalysis
conda activate mdanalysis
```

**Reply by hanahg** [reply]
Thank you very much

**Reply by IAlibay** [reply]
@hanahg Given that we've moved to github discussions, please disregard my comment about discord. This is a good place to workshop your installation issues.
